### PR TITLE
perf: vectorized image augs and a torch.compile knob for HPT and Pi

### PR DESCRIPTION
**Pull request by RyanPCo**
perf: vectorized image augs and a torch.compile knob for HPT and Pi

The 300M HPT train step was launch-bound, not compute-bound: ~14,600 CUDA
kernel launches per step, 230 ms of CPU against 111 ms of GPU. Two things
dominated, and both are fixed here.

PerSampleAugs called torchvision's ColorJitter once per image so each sample
drew its own factors -- 64 Python round trips a step, each launching a few
dozen kernels and copying its factors from the host. BatchedColorJitter draws
the factors as one on-device tensor and runs each adjustment once over the
batch; a degenerate factor range reproduces torchvision's scalar adjustment
exactly (tests/unit/test_image_augs.py). Anything it cannot batch still goes
through the per-image loop.

model.compile.enabled=true routes to <algo>.compile_for_training(), which uses
nn.Module.compile rather than torch.compile(module) so state_dict keys survive
-- the wrapper form renames every parameter _orig_mod.* and would break
checkpoints written before the flag. That only routes __call__, so HPT compiles
the trunk, the flow head's denoiser and the image encoders (the stems and heads
are reached through named methods), and PI's call site moves to __call__.

On one H100 at batch 64 and the 360x640 frames mecka decodes to: HPT 245 ->
173 ms from the augs, -> 109 ms compiled (2.24x). Pi at batch 16: 827 -> 588 ms
compiled (1.41x). In fp32 the compiled loss is bit-identical to eager
(bench/check_compile_parity.py).

Also here: the flow head's attentions pass need_weights=False so they dispatch
to scaled_dot_product_attention; frames decode to float32 instead of numpy's
float64 default, halving the collate and H2D traffic (338 -> 169 MB a camera a
batch, bit-identical); a per-camera per-step device->host sync is gone; and the
batch is no longer deep-copied every step for an OT loss that is off.

bench/README.md records the harnesses, the numbers, what was tried and did not
pay (persistent_workers, pin_memory, more workers, max-autotune), and two loose
ends -- ModelWrapper never receives enable_grad_norm, and the shared emimic/
venv is off-pin for transformers.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01VQa3o51gPcrK8N1LLoqL95

bench: verify the perf changes do not move accuracy

The speed commit changed three things that are not bit-identical to what they
replaced, and it only proved one of them. This adds the other two.

need_weights=False moves the flow head's attentions onto
scaled_dot_product_attention. check_attn_parity.py forces the old
materialized-softmax path back on and compares a full fp32 step: loss
197.009293 vs 197.009033 (1.3e-6 relative), 581 gradient tensors at min cosine
0.999999. Reduction order, nothing else.

The batched ColorJitter is the one real behaviour change -- it draws the op
order once per call rather than once per image, so a batch shares a
permutation. check_aug_distribution.py shows a single sample's marginal is
untouched (6400 samples a path, two-sample KS over eight per-sample
statistics, worst p = 0.61); only a training run can see the within-batch
correlation. ab_augs.sh runs both paths at two seeds -- 2000 steps, 32 train
episodes, held-out loss every 100 steps on 15 disjoint episodes, eval-mode augs
so the held-out pass measures what was learned rather than what was shown. The
batched arm is better at both seeds (-9.0%, -23.2%) and the 1.26x wall-clock
gap confirms the control arm really took the per-image path.

That rules out a regression; it is NOT evidence the rewrite helps. Two seeds
cannot carry that, and the arms consume different amounts of RNG, so each
arm/seed pair is effectively its own seed. bench/README.md says so where
someone would otherwise quote the number.

The control arm reuses the shipped fallback: PerSampleAugs keeps its per-image
loop for aug lists _vectorize declines, so stubbing _vectorize out runs the old
code rather than a reimplementation of it.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01VQa3o51gPcrK8N1LLoqL95

### CODE REVIEW

**Comment by RyanPCo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/676?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#680** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/680?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#681](https://github.com/GaTech-RL2/EgoVerse/pull/681) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/681?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#682](https://github.com/GaTech-RL2/EgoVerse/pull/682) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/682?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#679** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/679?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#678** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#677** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/677?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#676** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/676?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/676?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#674** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/674?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#673** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/673?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#672** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/672?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#671** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#670** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#669** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/669?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#668** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#667** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#666** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/666?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#665** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/665?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#664** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#646** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/646?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#647](https://github.com/GaTech-RL2/EgoVerse/pull/647) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/647?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#645** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/645?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#650** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/650?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#649** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/649?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#652** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/652?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#651** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/651?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#644** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/644?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#643** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/643?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#642** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/642?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#640** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/640?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#659** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/659?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#658** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/658?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#657** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/657?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#656** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/656?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#655** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/655?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#654** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/654?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#653** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/653?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#638** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/638?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#639](https://github.com/GaTech-RL2/EgoVerse/pull/639) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/639?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#637** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/637?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#636** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/636?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#634** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/634?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#635](https://github.com/GaTech-RL2/EgoVerse/pull/635) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/635?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#633** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/633?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#631** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/631?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#632](https://github.com/GaTech-RL2/EgoVerse/pull/632) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/632?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#630** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/630?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#628** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/628?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#629** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/629?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#626** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/626?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#625** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/625?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#624** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/624?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#623** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/623?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#622** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/622?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#627** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/627?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#621](https://github.com/GaTech-RL2/EgoVerse/pull/621) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/621?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#620** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/620?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#619** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
